qpmodel/optimizer.cs

```diff
@@ -50,8 +51,61 @@ public class Property { }
     public class PhysicProperty : Property
     {
         // ordering: the ordered expression and whether is descending
-        KeyValuePair<Expr, bool> ordering_;
-        KeyValuePair<Expr, DistributionType> distribution_;
+        List<KeyValuePair<Expr, bool>> ordering_;
```

Comment:
We may have COLLATE in future in sort clause, so sort is ORDER BY <expr> <desc?> <collation?>. You can have a refactoring patch to introduce a SortByExpr class grouping them together.